stdlib/public/core/UnsafePointer.swift

```diff
@@ -784,7 +784,9 @@ public struct UnsafeMutablePointer<Pointee>: _Pointer {
   ///   - count: The number of instances to move from `source` to this
   ///     pointer's memory. `count` must not be negative.
   @inlinable
-  public func moveInitialize(from source: UnsafeMutablePointer, count: Int) {
+  public func moveInitialize(
+    @_nonEphemeral from source: UnsafeMutablePointer, count: Int
```

Comment:
This and `moveAssign` can be ephemeral; the pointer argument isn't persisted.

Reply:
@jrose-apple While the pointer argument isn't persisted, passing a temporary pointer conversion is still unsound because we'll over-release the value. We could emit a more specialized diagnostic for this, but I didn't think it was worth implementing in this PR. What do you think?

Reply:
Hm, okay, I didn't think that through and that's evidence that it's best to leave it. People can explicitly use `withUnsafeMutableBufferPointer` or whatever if they want to get around this.